pyproject.toml

```diff
@@ -61,3 +62,13 @@ addopts = "--tb=short -s"
 # log_format = "%(asctime)s %(levelname)s %(message)s"
 # log_date_format = "%Y-%m-%d %H:%M:%S"
 log_cli = true # when true, messages are printed immediately
+
+[tool.ruff]
+exclude = [
+    "*.pyi"
```

Comment:
Is adding ruff directly related to this PR? If not, that's OK, but why ignore the pyi file?